datafusion/common/src/tree_node.rs

```diff
@@ -532,7 +532,7 @@ impl<T> Transformed<T> {
     }
 }
 
-/// Transformation helper to process tree nodes that are siblings.
+/// Transformation helper to process sequence of iterable tree nodes that are siblings.
```

Comment:
I found the reason / logic for continue / jump quite subtle when trying to make TreeNodeMutator. I have some suggested comments below to explain the rationale that might help

Reply:
Thanks @alamb, merged your suggestion.